### PR TITLE
feature(azure-iot-device): Added model_id to connection username

### PR DESCRIPTION
**Pull request by cartertinney**
* Added `model_id` config option
* Changed `model-id` username query parameter to use the value from the `model_id` config option
* Removed provisionary support for model ids provided via the `product_info` config option
* Removed the need for two different API versions for PNP / non-PNP

### CODE REVIEW

File: azure-iot-device/azure/iot/device/constant.py [hunk 1]
```diff
@@ -10,13 +10,9 @@
...
 IOTHUB_API_VERSION = "2019-10-01"
+IOTHUB_DIGITAL_TWIN_API_VERSION = "2020-09-30"
 PROVISIONING_API_VERSION = "2019-03-31"
 SECURITY_MESSAGE_INTERFACE_ID = "urn:azureiot:Security:SecurityAgent:1"
 TELEMETRY_MESSAGE_SIZE_LIMIT = 262144
 # The max keep alive is determined by the load balancer currently.
 MAX_KEEP_ALIVE_SECS = 1740
-# Everything in digital twin is defined here
-# as things are extremely dynamic and subject to sudden changes
-DIGITAL_TWIN_PREFIX = "dtmi"
```
Comment (cartertinney) [inline azure-iot-device/azure/iot/device/constant.py:20]: Removed these for consistency with how other query parameters are handled

File: azure-iot-device/azure/iot/device/iothub/pipeline/pipeline_stages_iothub_mqtt.py [hunk 2]
```diff
@@ -51,32 +51,33 @@ def _run_op(self, op):
...
             # See the repo wiki article for details:
             # https://github.com/Azure/azure-iot-sdk-python/wiki/URL-Encoding-(MQTT)
-            username = "{hostname}/{client_id}/?{query_params}".format(
+            username = "{hostname}/{client_id}/?{query_params}{product_info}".format(
                 hostname=self.pipeline_root.pipeline_configuration.hostname,
                 client_id=client_id,
                 query_params=version_compat.urlencode(
                     query_param_seq, quote_via=urllib.parse.quote
                 ),
+                product_info=urllib.parse.quote(custom_product_info, safe=""),
```
Comment (cartertinney) [inline azure-iot-device/azure/iot/device/iothub/pipeline/pipeline_stages_iothub_mqtt.py:80]: This is a clarity change - functionally it does the same thing it was always doing, but it's a lot more readable now, and results in less immediate questions, such as "wait, why was custom product info being appended to the end of the user agent query parameter on line 65????")

**Review by vishnureddy17 — APPROVED** [review]

**Review by BertKleewein — APPROVED** [review]
:shipit:

File: azure-iot-device/azure/iot/device/iothub/pipeline/pipeline_stages_iothub_mqtt.py [hunk 3]
```diff
@@ -51,32 +51,33 @@ def _run_op(self, op):
...
             custom_product_info = str(self.pipeline_root.pipeline_configuration.product_info)
-            if custom_product_info.startswith(
-                pkg_constant.DIGITAL_TWIN_PREFIX
-            ):  # Digital Twin Stuff
-                query_param_seq.append(("api-version", pkg_constant.DIGITAL_TWIN_API_VERSION))
+
+            if self.pipeline_root.pipeline_configuration.model_id is not None:
+                # Digital Twin Stuff
+                query_param_seq.append(
+                    ("api-version", pkg_constant.IOTHUB_DIGITAL_TWIN_API_VERSION)
```
Comment (BertKleewein) [inline azure-iot-device/azure/iot/device/iothub/pipeline/pipeline_stages_iothub_mqtt.py:58]: >IOTHUB_DIGITAL_TWIN_API_VERSION [](http://example.com/codeflow?start=49&length=31)

we may not need a specific API versoin for digital twin stuff now.

**Pull request merged**